app/controllers/api/v1/users/approvals_controller.rb

```diff
@@ -3,41 +3,41 @@ class Api::V1::Users::ApprovalsController < Api::ApiController
 
   acts_as_token_authentication_handler_for User, except: :create
 
-  before_action :check_user, only: :update
-
   def create
     resource_exists?(approvable_type, approvable)
     Approval.link(@user, approvable, approvable_type)
     accept_if_friend_request_or_adding_developer if req_from_coposition_app?
     approval = @user.approval_for(approvable)
-    @dev.notify_if_subscribed('new_approval', approval_zapier_data(approval))
+    @dev.notify_if_subscribed("new_approval", approval_zapier_data(approval))
     render json: approval
   end
 
   def update
-    approval = Approval.find_by(id: params[:id], user: @user)
-    return unless approval_exists? approval
-    approvable = approval.approvable
-    type = approval.approvable_type
-    Approval.accept(@user, approvable, type)
-    render json: approval.reload
-    approvable.notify_if_subscribed('new_approval', approval_zapier_data(approval)) if type == 'Developer'
+    result = ::Users::Approvals::UpdateApproval.call(current_user: @user, params: params)
+    if result.success?
+      render json: result.approval.reload
```

Comment:
it wouldn't

I guess we might use something like `render '...' and return ...` in this case